plugins/common/Distortion.hpp

```diff
@@ -89,34 +138,73 @@ class Distortion : public Effect
   {
     for (int i = 0; i < framesize; i++)
     {
-      float value = clipSample(input[i] * m_gain) * m_level;
-      value = m_lpf.getSample(value);
-      value = m_hpf.getSample(value);
+      // HPF in chain - determines amount of low-freqs sent to clipper block
+      // use F~140 Hz for full-range. F=720 for Tubescreamer type distorion
+      float value = m_hpf.getSample(value);
+      
+      // Stage 1 - first clipper (set to emulate op-amp clipping before diodes)
+      value = clipSample(input[i] * m_gain1); // clip signal
+      value = m_lpf1.getSample(value);   // filter out higher-order harmonics
+
+      // Stage 2 - diode clipper 
+      value = clipSample(input[i] * m_gain2) * m_level;
```

Comment:
I think this should also be value instead of input[i] if you want to cascade the clippers.  This line invalidates the first hpf and clipper